### PR TITLE
Post for building and releasing to aws with VSTS.

### PR DESCRIPTION
**Pull request by kevbite**

### CODE REVIEW

File: _posts/2018-08-13-build-and-releasing-with-vsts-for-multiple-aws-serverless-stacks.md [hunk 1]
```diff
@@ -0,0 +1,436 @@
...
+tags: [AWS, VSTS, Serverless, C#, CloudFormation, CI]
+description: A how to, for splitting up the build and release pipelines for AWS Serverless Applications in VSTS
+comments: true
+---
+
+With any project it's always good to be able to split up the build and the release stages of your application. This allows you to just keep propagating the same build artifacts across each environment so you can be confident that the code will work the same once it gets to production.
+
+## Prerequisites
+
+For this example we'll following to be install or setup:
```
Comment (sakisk) [inline _posts/2018-08-13-build-and-releasing-with-vsts-for-multiple-aws-serverless-stacks.md]: > For this example we'll following

we'll be following

File: _posts/2018-08-13-build-and-releasing-with-vsts-for-multiple-aws-serverless-stacks.md [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
+{
+    public static HttpClient HttpClient { get; } = new HttpClient();
+
+    public static string StackName { get; } = Environment.GetEnvironmentVariable("STACK_NAME");
+    public static string HttpEndpoint { get; } = Environment.GetEnvironmentVariable("HTTP_ENDPOINT");
+
+    public async Task FunctionHandler(S3Event evnt, ILambdaContext context)
+    {
+        var s3Event = evnt.Records?[0].S3;
+        if(s3Event == null)
```
Comment (sakisk) [inline _posts/2018-08-13-build-and-releasing-with-vsts-for-multiple-aws-serverless-stacks.md:57]:
```
if (s3Event == null)
  return;
```

File: _posts/2018-08-13-build-and-releasing-with-vsts-for-multiple-aws-serverless-stacks.md [hunk 3]
```diff
@@ -0,0 +1,436 @@
...
+
+![vsts-build-success-artifacts]
+
+![vsts-build-artifacts-explorer]
+
+## Release Pipeline
+
+Now we have our build artifacts ready to deploy, we will need to create a Release pipeline to deploy to AWS.
+
+However, before we get started on our pipeline, we need to Install the AWS VSTS Extensions and Configure a AWS Service Endpoint. We will not the full details of setting this up but you can follow the [Getting Started](https://docs.aws.amazon.com/vsts/latest/userguide/getting-started.html) section of the AWS VSTS documentation which will guide you through the process.
```
Comment (sakisk) [inline _posts/2018-08-13-build-and-releasing-with-vsts-for-multiple-aws-serverless-stacks.md]: > We will not the full details of setting this up but you can follow the

correct syntax

File: _posts/2018-08-13-build-and-releasing-with-vsts-for-multiple-aws-serverless-stacks.md [hunk 4]
```diff
@@ -0,0 +1,436 @@
...
+
+To start with we need to tell VSTS where to pull artifacts to use within our release pipeline. Select the _Add an artifact_ block and select the project where we build our artifacts within the last section. We'll just leave the settings as the default as this will work perfectly for our scenario.
+
+![vsts-release-add-an-artifact]
+
+#### Environments
+
+##### DevTest Environment
+
+Now let's create our first environment, we'll call this `DevTest` for the time being but feel free to name it corresponding to your stack that you will be creating.
```
Comment (sakisk) [inline _posts/2018-08-13-build-and-releasing-with-vsts-for-multiple-aws-serverless-stacks.md:283]: > ut feel free to name it corresponding to your stack

improve syntax

File: _posts/2018-08-13-build-and-releasing-with-vsts-for-multiple-aws-serverless-stacks.md [hunk 5]
```diff
@@ -0,0 +1,436 @@
...
+
+![vsts-release-pipeline-clone-devtest-environment]
+
+Once cloned, rename the environment to _AWS Production_.
+
+![vsts-release-pipeline-environments]
+
+We'll now have to alter our _Stack Name_ within our _AWS Lambda .NET Core Deployment_ task, we'll call this `Production-MyApp` and now we're ready to roll our application to production!
+
+Note: We could of also change our CloudFormation template parameters such as `HttpEndpoint` to point to another endpoint for production but for simplicity we'll keep them the same.
```
Comment (sakisk) [inline _posts/2018-08-13-build-and-releasing-with-vsts-for-multiple-aws-serverless-stacks.md]: > Note: We could of also change 

correct syntax

File: _posts/2018-08-13-build-and-releasing-with-vsts-for-multiple-aws-serverless-stacks.md [hunk 6]
```diff
@@ -0,0 +1,436 @@
...
+
+![vsts-release-pipeline-production-release]
+
+![aws-cloudformation-stacks]
+
+Now that everything is built you'll notice that we now have identical stacks between DevTest and Production. We can go upload another file in to our newly create production s3 bucket, and this will trigger off our production dotnet core lambda function.
+
+![requestbin-production-capture]
+
+Below you can also see the 2 lambda functions that were create for each stack by CloudFormation.
```
Comment (sakisk) [inline _posts/2018-08-13-build-and-releasing-with-vsts-for-multiple-aws-serverless-stacks.md]: > that were create 

created

File: _posts/2018-08-13-build-and-releasing-with-vsts-for-multiple-aws-serverless-stacks.md [hunk 7]
```diff
@@ -0,0 +1,436 @@
...
+
+![aws-functions]
+
+## Combined Power of VSTS and AWS CloudFormation
+
+As you can appreciate this gives us lots of power to allow VSTS to track our work items and also allows us to progress them in to each environments instead of the default way of just having a stack deployed on every build.
```
Comment (sakisk) [inline _posts/2018-08-13-build-and-releasing-with-vsts-for-multiple-aws-serverless-stacks.md]: > to each environments

enviroment

**Review by sakisk — COMMENTED** [review]
syntax and grammar checks mostly...a code suggestion as well...hate curly braces... ❤️ ruby

File: _posts/2018-08-13-build-and-releasing-with-vsts-for-multiple-aws-serverless-stacks.md (hunk 2 above)
Comment (kevbite) [reply _posts/2018-08-13-build-and-releasing-with-vsts-for-multiple-aws-serverless-stacks.md:57]: 💩

**Pull request merged**